lib/fluent/plugin/filter_prometheus.rb

```diff
@@ -22,6 +24,17 @@ def configure(conf)
       @metrics = Fluent::Plugin::Prometheus.parse_metrics_elements(conf, @registry, labels)
     end
 
+    def start
+      super
+      Fluent::Plugin::Prometheus.start_retention_threads(
```

Comment:
Is it really need to use thread?
I'm not sure but `timer_execute` (run it in plugin thread) might be enough to do it.